packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -522,13 +554,23 @@ export class GarbageCollector implements IGarbageCollector {
             // For new documents, GC is enabled by default. It can be explicitly disabled by setting the gcAllowed
             // flag in GC options to false.
             this.gcEnabled = this.gcOptions.gcAllowed !== false;
-            // The sweep phase has to be explicitly enabled by setting the sweepAllowed flag in GC options to true.
-            this.sweepEnabled = this.gcOptions.sweepAllowed === true;
+            this.maxSnapshotCacheDurationMs = createParams.snapshotCacheExpiryMs ?? "none";
+
+            // The sweep phase has to be explicitly enabled by setting the sweepAllowed flag in GC options to true,
+            // and Sweep cannot be enabled if there's no maxSnapshotCacheDurationMs
+            this.sweepEnabled =
+                this.gcOptions.sweepAllowed === true
+                && this.maxSnapshotCacheDurationMs !== "none";
 
             // Set the Session Expiry only if the flag is enabled and GC is enabled.
             if (this.mc.config.getBoolean(runSessionExpiryKey) && this.gcEnabled) {
                 this.sessionExpiryTimeoutMs = this.gcOptions.sessionExpiryTimeoutMs ?? defaultSessionExpiryDurationMs;
             }
+            // If we're using TestOverrides to achieve a short Sweep Timeout, lock the buffer to 1ms. Otherwise 1 day.
+            this.sweepTimeoutBufferMs =
+                (this.maxSnapshotCacheDurationMs === 0)
+                    ? 1
```

Comment:
Why is this 1?

Reply:
Just felt like it should be non-zero 🤷‍♂️